### ISSUE TITLE
Getting the OpenGL Error 1281 (Invalid Value) While in a Roofed Forest Biome

### ISSUE THREAD

**Issue by Coraline77447** [ferreusveritas/Dynamic-Trees#588]
Minecraft: 1.16.5
Forge: 1.16.5 36.1.32 (and 36.1.31)
OptiFine: OptiFine_1.16.5_HD_U_G8*
Dynamic Trees: DynamicTrees-1.16.5-0.10.0-Beta17
Dynamic Trees - Biomes O' Plenty: DynamicTreesBOP-1.16.5-2.0.4

Describe the bug:
I've been experiencing a OpenGl error whenever I'm in or around a roofed forest biome and I've tried to fix the error by updating both forge and Optifine but no luck...

To Reproduce:
Steps to reproduce the behavior: Use the latest version of Optifine or the latest pre version of Optifine with Dynamic Trees and go into a roofed forest biome and wait till your chat starts to say (OpenGL Error 1281), also have [Show OpenGL Errors] on.

Rendering Glitches:
Are you using OptiFine (or any equivalent mods that mess with rendering)?;
Yes, I am using Optifine.

Additional context:
*When I was using ForgeOptifine 1.16.5 in my launcher (TLauncher for modded Minecraft ONLY) I went from a roofed forest biome to a wetland biome from Biomes O' Plenty (while in creative mode) and that's when I'd started getting the OpenGL error 1281. 
I don't exactly know which version of Optifine it was but I bet it was a pre version. 
At the time of the first appearance of the error that my forge version wasn't at the latest version (36.1.31).
So I've seen this appear in forge 1.16.5-36.1.31 and forge 1.16.5-36.1.32.
Anyways, I'd updated both Forge and OptiFine to try to fix the OpenGL error.
But during my last play session I'd saw the same OpenGL error pop up while I was deep inside a roofed forest biome that I was in picking poppies...
At a certain distance in front of me the terrain started to flash/flicker on and off while the terrain was going invisible which made the trees seem like they were floating...
So I'd looked away from the direction that I saw this happen in, then I'd teleported back home from the roofed forest biome when I was done gathering poppies... 
Was in there to pick poppies to make dyes for my guns in a gun mod that I'm playing with to add camo/paint to them...
Right before I started seeing the terrain in front of me going invisible while flashing/flickering on and off I'd turned off [Show OpenGL Errors] in my settings....

Crashes:
The game hasn't crashed so this part of the report is irrelevant/not useful...

Log:
Here's the latest log:

[latest.log](https://github.com/ferreusveritas/DynamicTrees/files/6776302/latest.log)

Btw I couldn't make a paste in pastbin cause the log was too long/big and I'm still kind of new to GitHub so I have no choice to drop the log in this report even tho it's not a proper way to do it in your issue tracker....

**Comment by Harleyoc1** [comment]
I see no solid proof DT is causing this? Could you at least try taking out DT (on a test world) to see if you get the same error? I've certainly used OptiFine and DT together before in testing and have never seen any errors like this.

**Comment by Coraline77447** [comment]
I'll Test it out before I continue with my adventure in my current modded world.
I'll take screenshots of it accruing in chat since I'd didn't do that when I'd started experiencing the issue.
Also what other info do you need to figure out how to resolve this issue?
Since the game hasn't crashed from this issue so crash reports are useless...

UPDATE: I'd made a test world with only dynamic trees, optifine, and blame mod.
Went to a dark forest biome to see if I can get the OpenGL error to appear again and it hasn't, so the testing was inconclusive...
Btw what I've meant by roofed forest biome was dark forest biome.
I'm gonng to recreate my current modded world with biomes o plenty and the compatibility mod (Dynamic Trees - Biomes O' Plenty) and find the same dark forest biomes that I've discovered on my journey to see if I can get it to appear again.

If it doesn't then what I've realized while I was testing with only DT, Optifine and blame mod was that my current modded world which is called Coraline World 3: A.O.A was created before DT beta 17 and with a pre version of OptiFine_U_G8 on forge 36.1.31.
The problem probably started after I've updated DT to beta 17 or even beta 16 since those chunks were already explored.
But the thing is how was I at the village in a wetland biome that's right next to a dark forest biome when I first started experiencing this issue when I wasn't even there on world creation? 
I don't remember being there at all...
I had discovered that village in my minimap when I was trying (and still trying) to fine a dark forest village from Repurposed Structures for the dark oak sapling that I need in order to craft a realm stone in Advent of Ascension (Nevermine).....

Anyways, what I'm saying is that I didn't consider about the age of my world as of when it was created with older versions of DT, OptiFine but mostly DT since I've never discovered this issue before when I created that world and traversed through the nearest dark forest biome which is right next to the biome that I'd spawned in and currently live in which is a bayou biome from Biomes O' Plenty on a small island.

**Comment by Coraline77447** [comment]
UPDATE 2: I've recreated my current modded world but the world generation was different than my current world (Coraline World 3: A.O.A)...
Anyways, I've located and teleported to the closest dark forest biome and still no luck of making it reappear again which means that my second test was also inconclusive....

So I've came to the conclusion that what's causing this error is the fact that Coraline World 3: A.O.A is an old world that was created before DT beta 17, with a pre version of optifine...
When I'd updated the mod that's when the error possibly began to pop up without notice...

I'm NOT going to restart Coraline World 3: A.O.A because I've made so much progress already (haven't really made little to any actual progress in Advent of Ascension but in general I have made progress which I'm not willing to lose to the 1281 OpenGL error which I can just simply just ignore it)...

So I'll let you close this report, I also would like to apologize for wasting your time with this report that shouldn't have been created in the first place 'cause I didn't take the age of the world itself to consideration to the cause of the OpenGL error since the world was originally created with an older version of DT and optifine....

But if the world breaks or gets corrupted somehow in some way then I can load a backup since the game kept prompting me to create a backup and load or to continue without making a backup (I don't know why I'm not getting prompted to backup the world apoun loading it anymore but I'll make a backup before joining the world again just in case something bad happens during my game session that may be world breaking and I had always made backups for that world till the prompt stopped poping up apoun joining the world).

**Comment by Harleyoc1** [comment]
I'm failing to see how starting a world in an old version would cause rendering issues. It's still rendering the same blocks, it shouldn't make a difference that OptiFine may have rendered them differently when the world started (unless OptiFine are doing something very unusual, which admittedly is a possibility).

**Comment by Harleyoc1** [comment]
So in your tests you used the exact same set of mods with the exact same versions, correct?

**Comment by Stereo528** [comment]
Yea this doesn't seem like a DT issue, could be a resource pack issue as there are many resource packs enabled, and the OptiFine error is "Copy VBO Range", which I'm not sure what that means tbh, it could be related to Internal Shaders if those are enabled.
The trees floating is likely due to how DT might render the trees and aren't included in optifine's rendering mechanics for blocks.

In either case, I feel like this might be better for the OptiFine bug tracker: [sp614x/optifine](https://github.com/sp614x/optifine/issues) instead of here?

**Comment by Harleyoc1** [comment]
OK well I'll close this for now then. If you (or someone else) experiences this and can narrow it down to actually being caused by DT then please do comment.

**Comment by Coraline77447** [comment]
I've looked up what Copy VBO Range is or what VBO stands for and it stands for Vertex Buffer Object.
I've went further into my investigation about this which I went to https://www.minecraftforum.net/forums/support/java-edition-support/3079790-minecraft-opengl-error and the person who commented below said "Use Forge 36.0.42
and OptiFine HD U G7".
So what may be the main cause of this is OptiFine (probability: HIGH) and Forge 36.1.31 (currently using forge 36.1.58)(Probability: MEDIUM/LOW) which at the time was an older version of forge when forge 36.1.32 was the latest version of forge.
This issue may still continue to happen with OptiFine installed from Forge 36.1.31 to Forge 36.1.62 or whatever the latest version of forge is by now. 
Also I wasn't and still ain't using shader packs cause they consume too much ram and the fact that both Coraline World 3: A.O.A and my current world that I'm playing on have 50+ mods (current world has 70+) while working with limited amount of RAM (8GB, 4GB for the game and the rest for the PC itself).
The main resource pack that I'm using is Faithful x32 which I've just been thinking that DT may not work well with that pack but it's the only pack that I like and can use while Soartex Fanver(x64 only) (my most favorite resource pack) is a little too much and PureBDCraft as of x128 is way too much in modded....
